src/middlewared/middlewared/plugins/vm.py

```diff
@@ -2035,8 +2035,13 @@ class VMFSAttachmentDelegate(FSAttachmentDelegate):
 
     async def query(self, path, enabled):
         vms_attached = []
+        vms = {
+            vm['id']: vm for vm in await self.middleware.call(
+                'vm.query', [('status.state', '=' if enabled else '!=', 'RUNNING')]
+            )
+        }
         for device in await self.middleware.call('datastore.query', 'vm.device'):
-            if device['dtype'] not in ('DISK', 'RAW'):
+            if (device['dtype'] not in ('DISK', 'RAW')) or device['vm']['id'] in vms:
```

Comment:
`vms` contains list of VMs that are subject of our interest, right? (started if `enabled is True`, stopped otherwise) Why do you skip devices that are part of these VMs then?